### PR TITLE
drivers: caam: cache align Key blob modifier

### PR DESCRIPTION
**Pull request by sahilnxp**
An array was used as Key blob modifier, changed it cache aligned dynamic memory.

### CODE REVIEW

File: core/drivers/crypto/caam/caam_key.c [hunk 1]
```diff
@@ -743,6 +745,12 @@ enum caam_status caam_key_init(void)
 
 	assert(alloc_size <= CFG_CORE_BIGNUM_MAX_BITS);
 
+	key_blob_modifier = caam_calloc_align(KEY_BLOB_MODIFIER_SIZE);
+	if (!key_blob_modifier)
+		return CAAM_FAILURE;
+
+	memcpy(key_blob_modifier, KEY_BLOB_MODIFIER, sizeof(KEY_BLOB_MODIFIER));
```
Comment (jenswi-linaro) [inline core/drivers/crypto/caam/caam_key.c:752]: s/sizeof(KEY_BLOB_MODIFIER)/KEY_BLOB_MODIFIER_SIZE/

Comment (sahilnxp) [reply core/drivers/crypto/caam/caam_key.c]: Ok

**Comment by sahilnxp** [comment]
Pull request updated.

File: core/drivers/crypto/caam/caam_key.c [hunk 2]
```diff
@@ -743,6 +745,12 @@ enum caam_status caam_key_init(void)
 
 	assert(alloc_size <= CFG_CORE_BIGNUM_MAX_BITS);
 
+	key_blob_modifier = caam_calloc_align(KEY_BLOB_MODIFIER_SIZE);
+	if (!key_blob_modifier)
+		return CAAM_FAILURE;
+
+	memcpy(key_blob_modifier, KEY_BLOB_MODIFIER, KEY_BLOB_MODIFIER_SIZE);
```
Comment (etienne-lms) [inline core/drivers/crypto/caam/caam_key.c:752]: I suggest to add `static_assert(sizeof(KEY_BLOB_MODIFIER) >= KEY_BLOB_MODIFIER_SIZE)` for consistency.
Or, alternatively:`#define KEY_BLOB_MODIFIER_SIZE (sizeof(KEY_BLOB_MODIFIER) - 1)`.

Comment (sahilnxp) [reply core/drivers/crypto/caam/caam_key.c:753]: Ok.

**Review by jenswi-linaro — COMMENTED** [review]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by sahilnxp** [comment]
Thanks @jenswi-linaro  Tags updated.

**Pull request merged**